include/swift/Basic/OwnedString.h

```diff
@@ -37,6 +37,19 @@ enum class StringOwnership {
   Copied,
 };
 
+char *copyString(const char *s, size_t length) {
+  assert(s && "cannot copy a null string");
+  assert(length > 0 && "cannot copy an empty string");
+
+  char *data = static_cast<char *>(malloc(length + 1));
+  assert(data && "expected successful malloc");
+
+  memcpy(data, s, length);
+  data[length] = '\0';
+
+  return data;
+}
+
```

Comment:
Let's do the following, please.
- Change  `OwnedString(const char *Data, size_t Length, StringOwnership Ownership)` to:
  `OwnedString(const char *Data, size_t Length)`
  - assume `StringOwnership::Unowned`
- Move `copyString(const char *s, size_t length)` to `OwnedString OwnedString::copy()`
- Add *rigorous* testing for copying. This is an absolute must if we are going to be implementing the equivalent to `strndup`.